File: src/models/Notification.js
const mongoose = require("mongoose");
const { NOTIFICATION_TYPES } = require("../constants/roles");

const notificationSchema = new mongoose.Schema(
  {
    user: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true,
      index: true
    },
    type: {
      type: String,
      enum: Object.values(NOTIFICATION_TYPES),
      required: true
    },
    title: {
      type: String,
      required: true,
      trim: true
    },
    message: {
      type: String,
      required: true,
      trim: true
    },
    relatedBorrowRecord: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "BorrowRecord"
    },
    dueDate: {
      type: Date
    },
    isRead: {
      type: Boolean,
      default: false
    },
    readAt: {
      type: Date
    }
  },
  {
    timestamps: true
  }
);

module.exports = mongoose.model("Notification", notificationSchema);

File: java-backend/Notification.java
class Notification {
  String id;
  String userId;
  String type;
  String title;
  String message;
  boolean read;
  String relatedBorrowRecord;
  String dueDate;
  String createdAt;
  String updatedAt;

  Notification(String id, String userId, String type, String title, String message, boolean read, String relatedBorrowRecord, String dueDate, String createdAt) {
    this.id = id;
    this.userId = userId;
    this.type = type;
    this.title = title;
    this.message = message;
    this.read = read;
    this.relatedBorrowRecord = relatedBorrowRecord;
    this.dueDate = dueDate;
    this.createdAt = createdAt;
    this.updatedAt = createdAt;
  }
}
